tests/linearizability/failpoints.go

```diff
@@ -165,6 +171,42 @@ func (f goPanicFailpoint) pickMember(t *testing.T, clus *e2e.EtcdProcessCluster)
 	}
 }
 
+func (f goPanicFailpoint) Available(member e2e.EtcdProcess) bool {
+	fs, err := listFailpoints(member)
+	if err != nil {
+		return false
+	}
+	return f.available(fs)
+}
+
+func (f goPanicFailpoint) available(fs []string) bool {
+	for _, failpoint := range fs {
+		if f.failpoint == failpoint {
+			return true
+		}
+	}
+	return false
+}
+
+func listFailpoints(member e2e.EtcdProcess) ([]string, error) {
```

Comment:
Please consider to get all available failpoints only once in `TestMain`. Obviously it doesn't make sense to trigger a http request on each test case.

Reply:
We can only get the available failpoints after starting the cluster, so the other option is to cache the available failpoints so that they can be reused.

Reply:
its not a problem as we currently trigger one failpoint per cluster.

Reply:
I was talking about reuse the failpoints across test cases. But since it is possible that different test case might run test against different etcd versions, so it might not be correct.

Have you considered adding retry logic when getting an HTTP status code >= 500?

Reply:
For original question, each test should be independent so I don't want to reuse anything. 

As for checking 500, I prefer to check availability early and assume all errors are errors. I think this is less flake prone approach than trying to handle errors on request side.

Reply:
> I think this is less flake prone approach than trying to handle errors on request side.

Do not agree this. Since you already handle the retry mechanism at [linearizability_test.go#L88](https://github.com/etcd-io/etcd/blob/e73f55d4e94666c99558baa2fd4e365aeaca4dc4/tests/linearizability/linearizability_test.go#L88). So there is no need to retry in `listFailpoints`.